core/tauri-utils/src/assets.rs

```diff
@@ -163,6 +166,10 @@ impl Assets for EmbeddedAssets {
       .map(|a| Cow::Owned(a.to_vec()))
   }
 
+  fn iter(&self) -> Box<dyn Iterator<Item = (&&str, &&[u8])> + '_> {
+    Box::new(self.assets.into_iter())
```

Comment:
Do we really need the box here? can't we construct a new type `AssetEntries` that implements `Iterator`

Reply:
it needs to be `dyn` because we're in the context of a trait.. so it needs Box too to be sized. Writing another type doesn't help this. It's also the solution used for csp_hashes.

Reply:
If we return an `AssetsEntries` struct, we don't need to deal with `dyn` and can just implement `Iterator` on `AssetsEntries`
```rs
      fn iter<'a>(&'a self) -> AssetsEntries<'a, &'_ str, &'_ [u8]>;
```

This should also be done for `csp_hashes` as well but to avoid breaking changes, we can delay that change to v2

Reply:
but then the AssetEntries need to hold the iterator anyway, unless we force the type to be a HashMap which isn't a good idea really for a trait

Reply:
The only downside I see with this, is that `AssetsEntries` will need to use `phf::map::Entries` as a concrete type under the hood and now the `trait Assets` is not so generic and that makes me wonder, do we need this trait? seems like our usage of it, could be just replaced with the concrete `EmbeddedAssets` type

Reply:
for tests it's nice to have this trait

but we can also put the author @chippers on the line

Reply:
this is not a blocker btw, just would be a nice improvement for v2 (unless the trait is really required)

Reply:
Late on seeing this @amrbashir @lucasfernog-crabnebula , the asset trait allows people to implement their own asset handling, such as if they want to fetch them from disk or network. it's one of the original reasons it became possible to change the items in a `Context`